### ISSUE TITLE
track.getCapabilities is not available on FireFox.

### ISSUE THREAD

**Issue by atuttle** [zxing-js/browser#72]
When attempting to use this library (via [react-qr-reader][qr]) in Firefox, the error `track.getCapabilities is not a function` is thrown by this line:

https://github.com/zxing-js/browser/blob/eaefab44679fbcd4eb579f888905b6712018276f/src/readers/BrowserCodeReader.ts#L94

In my testing, `track.getCapabilities` is defined in Chrome, but not Firefox. For my purposes, it would be sufficient if the torch feature was assumed to be unavailable if `getCapabilities` is not defined.

[qr]: https://github.com/BlackBoxVision/react-qr-reader/

**Comment by Aspyryan** [comment]
Yea could we get an error thrown that we could catch and have proper handling for?

**Comment by ibrahimfarooq17** [comment]
It was fixed for me by adding wrapping the body of my `onResult` callback with a try catch block.

**Comment by gregg-cbs** [comment]
Yeah still getting this error on firefox, it does not seem to affect functionality:
![image](https://github.com/zxing-js/browser/assets/57114335/d0f4928d-959f-4d23-89ea-62ae03ed222e)
